grails-web-url-mappings/src/main/groovy/org/grails/web/mapping/DefaultUrlMappingEvaluator.java

```diff
@@ -563,16 +557,23 @@ private Object _invoke(String methodName, Object arg, Object delegate) {
         }
 
         private String establishFullURI(String uri, List<ConstrainedProperty> constrainedList) {
-            if (parentResources.isEmpty()) {
+            if (parentResources.isEmpty() || inGroupConstraints) {
```

Comment:
Make sure we don't attempt to re-write the URI when processing child constraints.